eventuate-cdc-data-producer-wrappers/src/main/java/io/eventuate/cdc/producer/wrappers/kafka/TopicPartitionSender.java

```diff
@@ -97,10 +74,41 @@ But other thread added message to queue (just after we got null message) and tri
     }
   }
 
+
+  private byte[] processBatchIfNecessary(TopicPartitionMessage message) {
+    if (!enableBatchProcessing) {
+      return message.getBody().getBytes(Charset.forName("UTF-8"));
+    }
+
+    List<TopicPartitionMessage> batch = new ArrayList<>();
+
+    EventuateKafkaMultiMessageConverter.MessageBuilder messageBuilder = new EventuateKafkaMultiMessageConverter.MessageBuilder(batchSize);
+
+    if (!messageBuilder.addMessage(new EventuateKafkaMultiMessageKeyValue(message.getKey(), message.getBody()))) {
+      throw new RuntimeException("Cannot send kafka message, payload is too heavy!");
+    }
+
+    while (true) {
+      TopicPartitionMessage messageForBatch = messages.peek();
+
+      if (messageForBatch != null && messageBuilder
+              .addMessage(new EventuateKafkaMultiMessageKeyValue(messageForBatch.getKey(), messageForBatch.getBody()))) {
+
+        batch.add(messages.poll());
+      }
+      else {
+        break;
+      }
+    }
+
+    message.setBatch(batch);
```

Comment:
Why is the first message treated differently?

Reply:
I will unify